connector-email/src/test/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilderRHCLOUD28631SingleEmailPerUserTest.java

```diff
@@ -0,0 +1,145 @@
+     * set. The fix consisted on aggregating the users set to reduce the number
+     * of emails being sent.
+     * @throws IOException if the different stubbed responses could not be
+     * loaded.
+     */
+    @Test
+    void testMultipleRecipientSettingsSameUserSendSingleEmail() throws Exception {
+        // Prepare the payload that we will simulate that we receive on the
+        // connector.
+        final Set<String> users = Set.of("foouser", "baruser");
+        final RecipientSettings recipientSettings = new RecipientSettings(
+            true,
+            true,
+            null,
+            users
+        );
+        final Set<String> users2 = Set.of("johndoe", "janedoe");
+
+        final RecipientSettings recipientSettings2 = new RecipientSettings(
+            true,
+            true,
+            null,
+            users2
+        );
+
+        // Create the exchange that we will use in this test.
+        final Exchange exchange = this.createExchangeWithBody("");
+
+        final List<RecipientSettings> recipientSettingsList = new ArrayList<>();
+        recipientSettingsList.add(recipientSettings);
+        recipientSettingsList.add(recipientSettings2);
+
+        final List<String> subscribers = List.of("user1", "user2");
+
+        final String emailBody = "emailBody";
+        final String emailSubject = "emailSubject";
+        exchange.setProperty(ExchangeProperty.RENDERED_BODY, emailBody);
+        exchange.setProperty(ExchangeProperty.RENDERED_SUBJECT, emailSubject);
+        exchange.setProperty(ExchangeProperty.RECIPIENT_SETTINGS, recipientSettingsList);
+        exchange.setProperty(ExchangeProperty.SUBSCRIBERS, subscribers);
+
+        // Prepare the stubbed response for the RBAC service.
+        final URL url = Resources.getResource("processors/rbac/rbacUsersResponse.json");
+        final String body = Resources.toString(url, StandardCharsets.UTF_8);
+
+        // Mock the endpoints so that the whole flow is followed.
+        AdviceWith.adviceWith(this.context, Routes.FETCH_USERS_RBAC, a -> {
+            a.weaveByToUri(this.emailConnectorConfig.getRbacURL()).replace().setBody(new ConstantExpression(body));
+        });
+
+        // Make sure we replace the BOP endpoint too in order to check the
+        // usernames we are sending.
+        AdviceWith.adviceWith(this.context, Routes.SEND_EMAIL_BOP, a -> {
+            a.weaveByToUri(String.format("%s*", this.emailConnectorConfig.getBopURL())).replace().to("mock:bopendpoint");
+            a.mockEndpointsAndSkip(String.format("direct:%s", ConnectorToEngineRouteBuilder.SUCCESS));
+        });
+
+        // Set up the expected exchanges count that the success endpoint should
+        // receive at the end of the test.
+        final MockEndpoint successEndpoint = this.getMockEndpoint(String.format("mock:direct:%s", ConnectorToEngineRouteBuilder.SUCCESS));
+        successEndpoint.expectedMessageCount(4);
```

Comment:
The `engine` should only receive one Kafka message in the `fromcamel` topic, so this should be:
```suggestion
        successEndpoint.expectedMessageCount(1);
```

Reply:
Yes!